lib/data-types.js

```diff
@@ -267,7 +267,7 @@ var INTEGER = NUMBER.inherits(function(length) {
 
 INTEGER.prototype.key = INTEGER.key = 'INTEGER';
 INTEGER.prototype.validate = function(value) {
-  if (!Validator.isInt(value)) {
+  if (_.isString(value) && !Validator.isInt(value) || !_.isString(value) && !_.isInteger(value)) {
```

Comment:
In that case I think we can do the same regarding other number-like validations and also the boolean one. WDYT?